src/middleware/common.middleware.js

```diff
@@ -88,11 +94,300 @@ export function validateQueryParamsFn (req, res, next) {
   }
 }
 
+export const fetchSpecification = fetchOne({
+  query: ({ req }) => `select * from specification WHERE specification = '${req.dataset.collection}'`,
+  result: 'specification'
+})
+
+export const pullOutDatasetSpecification = (req, res, next) => {
+  const { specification } = req
+  const collectionSpecifications = JSON.parse(specification.json)
+  const datasetSpecification = collectionSpecifications.find((spec) => spec.dataset === req.dataset.dataset)
+  req.specification = datasetSpecification
+  next()
+}
+
+/**
+ *
+ * Middleware. Updates `req` with `issueEntitiesCount` which is the count of entities that have issues.
+ *
+ * Requires `req.resource.resource`
+ *
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next
+ */
+export async function fetchIssueEntitiesCount (req, res, next) {
+  const { dataset: datasetId, issue_type: issueType, issue_field: issueField } = req.params
+  const { resource: resourceId } = req.resource
+  console.assert(resourceId, 'missng resource id')
```

Comment:
Need to remove

Reply:
if intended - then we need to handle what should happen if resource doesn't exist

Reply:
also typo